sources/packages/backend/apps/api/src/route-controllers/confirmation-of-enrollment/_tests_/e2e/confirmation-of-enrollment.institutions.confirmEnrollment.e2e-spec.ts

```diff
@@ -0,0 +1,298 @@
+import { HttpStatus, INestApplication } from "@nestjs/common";
+import * as request from "supertest";
+import { DataSource, Repository } from "typeorm";
+import {
+  authorizeUserTokenForLocation,
+  BEARER_AUTH_TYPE,
+  createTestingAppModule,
+  getAuthRelatedEntities,
+  getInstitutionToken,
+  InstitutionTokenTypes,
+} from "../../../../testHelpers";
+import {
+  createFakeDisbursementValue,
+  createFakeInstitutionLocation,
+} from "@sims/test-utils";
+import { saveFakeApplicationCOE } from "@sims/test-utils/factories/confirmation-of-enrollment";
+import {
+  Application,
+  ApplicationStatus,
+  DisbursementSchedule,
+  DisbursementValueType,
+  EducationProgramOffering,
+  Institution,
+  InstitutionLocation,
+} from "@sims/sims-db";
+import {
+  COE_WINDOW,
+  MONEY_VALUE_FOR_UNKNOWN_MAX_VALUE,
+} from "../../../../utilities";
+import { addDays, getISODateOnlyString } from "@sims/utilities";
+import { ClientTypeBaseRoute } from "../../../../types";
+
+describe(`${ClientTypeBaseRoute.Institution}-ConfirmationOfEnrollmentInstitutionsController(e2e)-confirmEnrollment`, () => {
+  let app: INestApplication;
+  let appDataSource: DataSource;
+  let applicationRepo: Repository<Application>;
+  let disbursementScheduleRepo: Repository<DisbursementSchedule>;
+  let offeringRepo: Repository<EducationProgramOffering>;
+  let collegeC: Institution;
+  let collegeCLocation: InstitutionLocation;
+
+  beforeAll(async () => {
+    const { nestApplication, dataSource } = await createTestingAppModule();
+    app = nestApplication;
+    appDataSource = dataSource;
+    applicationRepo = dataSource.getRepository(Application);
+    disbursementScheduleRepo = dataSource.getRepository(DisbursementSchedule);
+    offeringRepo = dataSource.getRepository(EducationProgramOffering);
+
+    const { institution } = await getAuthRelatedEntities(
+      appDataSource,
+      InstitutionTokenTypes.CollegeCUser,
+    );
+    collegeC = institution;
+    collegeCLocation = createFakeInstitutionLocation(collegeC);
+    await authorizeUserTokenForLocation(
+      appDataSource,
+      InstitutionTokenTypes.CollegeCUser,
+      collegeCLocation,
+    );
+  });
+
+  it("Should allow the COE confirmation when the application is on Enrolment status and all the conditions are fulfilled.", async () => {
+    // Arrange
+    const application = await saveFakeApplicationCOE(appDataSource, {
+      institution: collegeC,
+      institutionLocation: collegeCLocation,
+    });
+    application.applicationStatus = ApplicationStatus.Enrolment;
```

Comment:
saveFakeApplicationCOE() creates the application in Enrolment status.

Reply:
Yes, it does now. I will change this.